reference-guide/configuration.adoc

```diff
@@ -1,4 +1,349 @@
+|`false`
+
+a|kogito.events.processdefinitions.enabled
+a|Toggle to enable publishing processes definition events.
+|boolean
+|`true`
+
+a|kogito.events.processdefinitions.errors.propagate
+a|Toggle to propagate errors for process definition emitter.
+|boolean
+|`false`
+
+a|kogito.events.usertasks.enabled
+a|Toggle to enable publishing user task instances events.
+|boolean
+|`true`
+
+a|kogito.events.usertasks.errors.propagate
+a|Toggle to propagate errors for user tasks.
+|boolean
+|`false`
+
+a|kogito.devservices.port
+a|Optional port the dev service will listen to.
+If not defined, 8180 will be used.
+|int
+|`8180`
+
+a|kogito.devservices.enabled
+a|Toggle to enable DevServices for Kogito.
+|boolean
+|true
+
+a|kogito.devservices.image-name
+a|The Data Index image to use.
+|string
+|
+
+a|kogito.devservices.shared
+a|Toggle to enable sharing of Quarkus Dev Services.
+When shared, Quarkus looks for running containers using label-based service discovery.
+If a matching container is found, it is used, so a second one is not started.
+Otherwise, Dev Services for a new Data Index instance is started.
+The discovery uses the `kogito-dev-service-data-index` label.
+The value is configured using the `service-name` property.
+Container sharing is only used in dev mode.
+|boolean
+|`true`
+
+a|kogito.devservices.service-name
+a|The value of the `kogito-dev-service-data-index` label attached to the started container.
+This property is used when `shared` is set to `true`.
+In this case, before starting a container, Dev Services looks for a container with the `kogito-dev-service-data-index` label set to the configured value.
+If found, it will use this container instead of starting a new one.
+Otherwise, it starts a new container with the `kogito-dev-service-data-index` label set to the specified value.
+|string
+|`kogito-data-index`
+
+a|kogito.messaging.as-cloudevents
+a|Toggle used to enable Cloud Events.
+|boolean
+|`true`
+
+a|kogito.jackson.fail-on-empty-bean
+a|Toggle if Jackson will fail on an empty bean.
+|boolean
+|`false`
+
+a|kogito.addon.messaging.incoming.defaultName
+a|Incoming channel default name.
+|string
+|`kogito_incoming_stream`
+
+a|kogito.addon.messaging.outgoing.defaultName
+a|Outgoing channel default name.
+|string
+|`kogito_outgoing_stream`
+
+a|kogito.codegen.ignoreHiddenFiles
+a|Toggle to ignore hidden files when collecting files for code generation.
+|boolean
+|`true`
+
+a|kogito.codegen.process.failOnError
+a|Toggle to fail when there are parsing/validation errors of process assets.
+|boolean
+|`true`
+
+a|kogito.service.url
+a|The service URL needed to connect to the runtime endpoint from outside the service.
+|string
+|
+
+a|kogito.security.auth.enabled
+a|Toggle Enabling the application security context when resolving current User Identity.
+|boolean
+a|false
+
+a|kogito.security.auth.impersonation.allowed-for-roles
+a|Comma-separated list of roles that allow identity impersonation when resolving the actual User Identity.
```

Comment:
Traditionally User Tasks REST Endpoints used the user/group query params to generate the user identity (username and roles) that is performing a task operation (transtion, add attachment/comment, modify inputs/outputs...) 

This property works in conjunction with `kie.security.auth.enabled`. If auth is enabled, we are overriding the previous mechanism to use the info from the security context to generate the user identity instead of using the query params.

However, there are some times it is possible that a `super user` has to perform an operation `impersonating` another user (imagine an admin completing a blocked task in behalf of another user). This property allows to configure a comma separated list of roles that are allowed to impersonate other users. 
So, when performing a task operation (with auth enabled) and the user in the app security context belongs to any of the impersonator roles, in this case we will generate the user Identity based on the query params.


For example a good value would be:
```
kogito.security.auth.impersonation.allowed-for-roles=manager,task-admin
``` 
This property has no default value, so no default impersonator groups are defined, this is totally up to the customer to configure it with the roles they want.

Reply:
Thanks @pefernan 
Same comment as https://github.com/IBM/bamoe-docs/pull/165#discussion_r1848181424

Reply:
@pefernan Excellent explanation! It will go into the next commit.